File: src/ts/magma/app/compile/type/TemplateType.ts
import { Type } from "../../../../magma/app/compile/type/Type";
import { Iterable } from "../../../../magma/api/collect/list/Iterable";
import { Merger } from "../../../../magma/app/compile/merge/Merger";
import { ValueMerger } from "../../../../magma/app/compile/merge/ValueMerger";
export class TemplateType implements Type {
	base: string;
	args: Iterable<string>;
	constructor (base: string, args: Iterable<string>) {
		this.base = base;
		this.args = args;
	}
	generate(): string {
		return this.base + "<" + Merger.generateAll(this.args, new ValueMerger()) + ">"/*unknown*/;
	}
	isFunctional(): boolean {
		return false/*unknown*/;
	}
	isVar(): boolean {
		return false/*unknown*/;
	}
	generateBeforeName(): string {
		return ""/*unknown*/;
	}
	generateSimple(): string {
		return this.base/*unknown*/;
	}
}

File: src/java/magma/app/compile/type/TemplateType.java
package magma.app.compile.type;

import magma.api.collect.list.Iterable;
import magma.app.compile.merge.Merger;
import magma.app.compile.merge.ValueMerger;

public record TemplateType(String base, Iterable<String> args) implements Type {
    public String generateType() {
        return this.base + "<" + Merger.generateAll(this.args, new ValueMerger()) + ">";
    }

    public boolean isFunctional() {
        return false;
    }

    public boolean isVar() {
        return false;
    }

    public String generateBeforeName() {
        return "";
    }

    public String generateSimple() {
        return this.base;
    }

    public boolean is(String type) {
        return "template".equals(type);
    }
}
